src/deepsparse/pipeline.py

```diff
@@ -405,13 +413,25 @@ def __call__(self, *args, **kwargs):
             inference_state = InferenceState()
             inference_state.create_state({})
 
-        kwargs["inference_state"] = inference_state
+            timer = self.timer_manager.get_new_timer()
+            inference_state.set_timer(timer)
 
         next_call = self.run
         if self.middleware_manager is not None:
+            # make next calls to be middlewares if any
             next_call = self.middleware_manager.build_middleware_stack(next_call)
 
-        return next_call(*args, **kwargs)
+        kwargs["inference_state"] = inference_state
+        # add name for timer measurements key
+        kwargs["name"] = "total"
```

Comment:
if a pipeline is nested we will get conflicts on the `"total"` name - need a way to resolve this (ie if timer already exists in inference state we know its a nested pipeline)